### DISCUSSION TITLE
Reconciliation after error state

### DISCUSSION THREAD

**Discussion by mpreu** [keycloak/keycloak#12108] [Archive]
Hi,

it seems that the operator does not requeue a CR anymore after it entered an error state. This is a bit inconvenient if e.g. one tries to instantiate a `kustomize` Keycloak configuration along with a database deployment:

1. Run `kustomize`
2. Postgres startup (Crunchy Postgres Operator) / Keycloak startup
3. Keyloak fails because the postgres secret is not populated in time
4. Postgres startup finishes, but Keycloak CR reconciling has now to be triggered manually

In this example, Keycloak OP fails with:

> Error performing operations:
> Secret xxx not found

The time frame for the availability of the referenced secret is just some seconds (maybe up to one minute) in the given example. 

One can argue, that such a configuration has to be splitted to allow postgres to be there first. But for development use cases or automation via ArgoCD I could imagine the reconciliation logic (or just timeout values / requeue time values) to be a bit more forgiving. 

I could also imagine to not totally end the reconciliation after an error (assuming this is the current behavior) to allow automatic error resolution for similar cases.

**Comment by andreaTP** [comment]
Hi @mpreu , thanks for checking out the new operator!

I agree that the current timeout/retry reschedule is a little "strict".

Until we come up with a more "structural" fix you can play with those settings:
https://quarkiverse.github.io/quarkiverse-docs/quarkus-operator-sdk/dev/index.html#quarkus-operator-sdk_quarkus.operator-sdk.controllers.-controllers-.retry.max-attempts (and following) passing them as environment variables [here](https://github.com/keycloak/keycloak-k8s-resources/blob/153d9308503380fa076780a01669e4c9751a0c69/kubernetes/kubernetes.yml#L201).
Please note that you would need to encode those parameters correctly, it should be something like:
`QUARKUS_OPERATOR_SDK_CONTROLLERS_CONTROLLERS_RETRY_MAX_ATTEMPTS`
(beware of typos!) but I haven't test it myself ...

I'm extremely interested in the result of these experiments!
